modules/standard/fun/fun_controller.py

```diff
@@ -27,44 +27,44 @@ def start(self):
     @command(command="beer", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random beer message")
     def beer_command(self, request, item_id):
-        return self.get_fun_message("beer", item_id)
+        return self.get_fun_message("beer", request, item_id)
 
     @command(command="brain", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random pinky and the brain quote")
     def brain_command(self, request, item_id):
-        return self.get_fun_message("brain", item_id)
+        return self.get_fun_message("brain", request, item_id)
 
     @command(command="chuck", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random Chuck Norris joke")
     def chuck_command(self, request, item_id):
-        return self.get_fun_message("chuck", item_id)
+        return self.get_fun_message("chuck", request, item_id)
 
     @command(command="cybor", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random cybor message Caution: HOT!")
     def cybor_command(self, request, item_id):
-        return self.get_fun_message("cybor", item_id)
+        return self.get_fun_message("cybor", request, item_id)
 
     @command(command="dwight", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random Dwight quote")
     def dwight_command(self, request, item_id):
-        return self.get_fun_message("dwight", item_id)
+        return self.get_fun_message("dwight", request, item_id)
 
     @command(command="fc", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random FC quote")
     def fc_command(self, request, item_id):
-        return self.get_fun_message("fc", item_id)
+        return self.get_fun_message("fc", request, item_id)
 
     @command(command="homer", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random homer quote")
     def homer_command(self, request, item_id):
-        return self.get_fun_message("homer", item_id)
+        return self.get_fun_message("homer", request, item_id)
 
     @command(command="pirates", params=[Int("item_id", is_optional=True)],
              access_level="member", description="Shows a random Pirates of the Caribbean quote")
     def pirates_command(self, request, item_id):
-        return self.get_fun_message("pirates", item_id)
+        return self.get_fun_message("pirates", request, item_id)
 
-    def get_fun_message(self, quote_type, number):
+    def get_fun_message(self, quote_type, request, number):
         if number is None:
             number = -1
         data = self.db.query("SELECT f.content FROM fun f WHERE type = ?", [str(quote_type)])
```

Comment:
If you add an id to the fun table, you should be able to avoid having to fetch all the quotes, when you really only want one.

Reply:
Thanks, this did need some work.

I changed the sql scripts and method so that:
- when **no** item id is specified; 
  - first query type row _count_
  - after that query a random row
- when item id **is** specified 
  - directly fetch that row